File: Problems/1030.py

class Solution(object):
    def allCellsDistOrder(self, rows, cols, rCenter, cCenter):
        """
        :type rows: int
        :type cols: int
        :type rCenter: int
        :type cCenter: int
        :rtype: List[List[int]]
        """
        answers = []

        for r in range(rows):
            for c in range(cols):
                answers.append((r, c))
        return sorted(answers, key=lambda x: (abs(rCenter - x[0]) + abs(cCenter - x[1])))
        

File: Problems/1030.scala
object Solution {
    def allCellsDistOrder(rows: Int, cols: Int, rCenter: Int, cCenter: Int): Array[Array[Int]] = {
        val temp: Array[Array[(Int, (Int, Int))]] = (for (
            i <- 0 to (rows - 1)
        ) yield (for (j <- 0 to (cols - 1))
         yield { 
                ((i - rCenter).abs + (j - cCenter).abs, (i, j))
            }).toArray).toArray
        
        val so: Array[(Int, (Int, Int))] = temp.flatten.sortWith((a, b) => a._1 < b._1)
        (for ((_, j: (Int, Int)) <- so) yield {
            Array[Int](j._1, j._2)
        }).toArray
    }
}